platform/macos/display_server_macos.mm

```diff
@@ -160,6 +160,7 @@
 							  defer:NO];
 		ERR_FAIL_NULL_V_MSG(wd.window_object, INVALID_WINDOW_ID, "Can't create a window");
 		[wd.window_object setWindowID:window_id_counter];
+		[wd.window_object setReleasedWhenClosed:NO];
```

Comment:
With autorelease pools, objects are getting freed properly, and this results in a crash if not set.